lib/mrtrix3/app.py

```diff
@@ -1101,6 +1101,88 @@ def _is_option_group(self, group):
            not group == self._positionals and \
            group.title not in ( 'options', 'optional arguments' )
 
+  class TypeBoolean:
+    def __call__(self, input_value):
+      true_var = "True"
+      false_var = "False"
+      converted_value = None
+      if input_value.lower() == true_var.lower():
+        converted_value = True
+      elif input_value.lower() == false_var.lower():
+        converted_value = False
+      else:
+        raise argparse.ArgumentTypeError('Entered value is not of type boolean')
+      return converted_value
+
+  class TypeIntegerSequence:
+    def __call__(self, input_value):
+      seq_elements = input_value.split(',')
+      int_list = []
+      for i in seq_elements:
+        try:
+          converted_value = int(i)
+          int_list.append(converted_value)
+        except:
+          raise argparse.ArgumentTypeError('Entered value is not an integer sequence')
```

Comment:
Good opportunity to use Python list comprehension.

Reply:
Also do not want to use all-encompassing exception catches; ideally only catch the type of exception that you *a priori* expect to be thrown if the user specifies something that isn't compatible with the expected type.

Reply:
Changes made in commit: https://github.com/ankitasanil/mrtrix3/pull/1/commits/54bca482ace8b52acd93b374b184cbd99244cbb9